### PR TITLE
Views features

### PR DESCRIPTION
**Pull request by mohashyne**
# New  features Added

### In this pull request, I've implemented the Views for the Blog App, aligning them with existing Models and Controllers. I've met all project specifications:

In terms of design:

    Implemented wireframes provided.
    Prioritized functionality over styling.
    Detailed analysis of wireframes for element identification and user navigation paths.

Implemented Views:

    All Users page
    Single User page
    All Posts by a User page
    Single Post page

Implementation Details:

    Utilized existing methods for data processing in models.
    Added sample data to the database.

Changes applied:
    Incorporated HTML and ERB code for content presentation.
    Integrated Blog App model functionalities for data representation.
    Ensured conformity to wireframe design in layout and elements.
    Validated smooth user navigation across pages.avigation between pages.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -1 +1,13 @@
...
+<h2>Total Comments</h2>
+<div>
+  <div class="user-card left">
+    <h3><%= @post.title %> by <%= @user.name %></h3>
+    <div class="right">
+        <span class="span-post">Comments: <%= @post.comments.count %></span>
+        <span class="span-post">Likes: <%= @post.likes.count %></span>
+      </div>
+      <br>
+    <p><%= @post.text %></p>
```
Comment (joel-kalema) [inline app/views/posts/show.html.erb]: - [x] Good job working on views, however, make sure to follow the given layout for the post show. Kindly display all comments for the post show 👍 

> | | template | | Your project | |
> |-|---------|-|--------|-|
> | |blog_single_post| |Screenshot 2023-10-05 171019| |

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @mohashyne 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 

- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [x] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mohashyne** [comment]
All changes required has been resolved

**Review by KDlamini — APPROVED** [review]
### Hi @mohashyne 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for the Blog app views. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞

✔️  Good PR title and description
✔️  Good and professional README documentation
✔️  Users controller `#index` action renders all users to their corresponding Views index page
✔️  `See all Post` link opens the user's post index as displayed on the wireframe
✔️  The `recent_posts` and `recent_comments` methods are utilized

### To Improve 🛠

**[Commits]: ** To follow the commit messages best practices, please ensure consistency in the way you write commit messages. It is good practice to start each commit message with a capital letter like you would any sentence.

# Status: Approved ✔️ 

great-work-approved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**